buildspec_integ.yml

```diff
@@ -6,54 +6,46 @@ phases:
-          # Enforce STS regional endpoints
-          export AWS_STS_REGIONAL_ENDPOINTS=regional
-          export CW_INTEG_VALIDATOR_IMAGE_BASE=${AWS_ACCOUNT}.dkr.ecr.us-west-2.amazonaws.com/cw-integ-validator
-          export S3_INTEG_VALIDATOR_IMAGE_BASE=${AWS_ACCOUNT}.dkr.ecr.us-west-2.amazonaws.com/s3-integ-validator
-          # Get the default credentials and set as environment variables
-          CREDS=`curl 169.254.170.2$AWS_CONTAINER_CREDENTIALS_RELATIVE_URI`
-          export AWS_ACCESS_KEY_ID=`echo $CREDS | jq -r .AccessKeyId`
-          export AWS_SECRET_ACCESS_KEY=`echo $CREDS | jq -r .SecretAccessKey`
-          export AWS_SESSION_TOKEN=`echo $CREDS | jq -r .Token`
-
-          # Pull the image that we built and pushed in the `Build` stage
-          aws ecr get-login-password --region ${AWS_REGION} | docker login --username AWS --password-stdin ${AWS_ACCOUNT}.dkr.ecr.${AWS_REGION}.amazonaws.com
-          docker pull ${AWS_ACCOUNT}.dkr.ecr.${AWS_REGION}.amazonaws.com/amazon/aws-for-fluent-bit-test:latest
-          docker tag ${AWS_ACCOUNT}.dkr.ecr.${AWS_REGION}.amazonaws.com/amazon/aws-for-fluent-bit-test amazon/aws-for-fluent-bit:latest
-
-          # List the images to do a double check
-          docker images
-
-          # Command to run the integration test
-          make integ
-          
-        elif [ "$BUILD_VERSION" = "3" ]; then
-          echo "Running integration tests for AWS for Fluent Bit version 3 (no-op for now)"
-          echo "Version 3 integration test logic will be implemented later"
-          # Placeholder for future BUILD_VERSION=3 implementation
-          
-        else
+        # Validate BUILD_VERSION
+        if [ "$BUILD_VERSION" != "2" ] && [ "$BUILD_VERSION" != "3" ]; then
           echo "Unsupported BUILD_VERSION: $BUILD_VERSION"
           echo "Supported versions are: 2, 3"
           exit 1
         fi
+      - |
+        # Enforce STS regional endpoints and set validator images
+        export AWS_STS_REGIONAL_ENDPOINTS=regional
+        export CW_INTEG_VALIDATOR_IMAGE_BASE=${AWS_ACCOUNT}.dkr.ecr.us-west-2.amazonaws.com/cw-integ-validator
+        export S3_INTEG_VALIDATOR_IMAGE_BASE=${AWS_ACCOUNT}.dkr.ecr.us-west-2.amazonaws.com/s3-integ-validator
+      - |
+        # Get the default credentials and set as environment variables
+        CREDS=$(curl 169.254.170.2$AWS_CONTAINER_CREDENTIALS_RELATIVE_URI)
+        export AWS_ACCESS_KEY_ID=$(echo $CREDS | jq -r .AccessKeyId)
+        export AWS_SECRET_ACCESS_KEY=$(echo $CREDS | jq -r .SecretAccessKey)
+        export AWS_SESSION_TOKEN=$(echo $CREDS | jq -r .Token)
+      - aws ecr get-login-password --region ${AWS_REGION} | docker login --username AWS --password-stdin ${AWS_ACCOUNT}.dkr.ecr.${AWS_REGION}.amazonaws.com
+      - |
+        # Set image tag based on BUILD_VERSION
+        if [ "$BUILD_VERSION" = "2" ]; then
+          IMAGE_TAG="latest"
+        elif [ "$BUILD_VERSION" = "3" ]; then
+          IMAGE_TAG="latest-3"
```

Comment:
This is the tag for test image pushed during [build.sh](https://github.com/aws/aws-for-fluent-bit/blob/2cb113e145586f30d8ac9768f88612e738568d95/scripts/build.sh#L130)